File: java/terminal/terminal-server/src/main/java/com/github/fernthedev/lightchat/server/terminal/command/TabExecutor.java
package com.github.fernthedev.lightchat.server.terminal.command;

import com.github.fernthedev.lightchat.server.SenderInterface;

import java.util.LinkedList;
import java.util.List;

public interface TabExecutor {

    /**
     * Returns a list of completions based on the arguments given
     * @param sender
     * @param args
     * @return
     */
    List<String> getCompletions(SenderInterface sender, LinkedList<String> args);


}

File: java/terminal/terminal-server/src/main/java/com/github/fernthedev/lightchat/server/terminal/command/TabExecutor.kt
package com.github.fernthedev.lightchat.server.terminal.command

import com.github.fernthedev.lightchat.server.SenderInterface
import java.util.*

interface TabExecutor {
    /**
     * Returns a list of completions based on the arguments given
     * @param sender
     * @param args
     * @return
     */
    fun getCompletions(sender: SenderInterface, args: Deque<String>): List<String>
}